### ISSUE TITLE
TCP socket server / faster telnet

### ISSUE THREAD

**Issue by narfdotpl** [arduino-libraries/UnoWiFi-Developer-Edition-Lib#7]
Helllo, @sergiotomasello 👋 

I want to communicate with Arduino Uno WiFi over a socket to control a servo. Unfortunately, I couldn't find an Uno WiFi example using a TCP socket server. I've tried sending servo positions to port 23 but servo movement is jerky — as if there was a `delay` after each `servo.write` call. Is there a way to have "faster telnet" or to open a socket on a different port?

Here's my Arduino code:

```arduino
#include <Servo.h>
#include <UnoWiFiDevEd.h>


const int servoPin = 9;
Servo servo;


void setup() {
    Wifi.begin();
    servo.attach(servoPin);
}


void loop() {
    while (Wifi.available()) {
        servo.write(Wifi.read());
    }
}
```

I posted more details in a [question on Stack Exchange](http://arduino.stackexchange.com/questions/35682/how-to-run-tcp-socket-server-on-arduino-uno-wifi). (I didn't want to copy-paste everything here.)

Thank you in advance. 🙇 

— Maciej

**Comment by sergiotomasello** [comment]
hi @narfdotpl i'm quite sure it's due to the baud rate between esp8266 and microcontroller (328P) and... let me check

**Comment by narfdotpl** [comment]
I still don't know why it works so well on Yún. Maybe it has a better antenna? Anyway, I've returned Uno WiFi and got a SparkFun ESP32 Thing. Unfortunately, it has the same problem as Uno WiFi. To fix it I've switched from TCP to UDP. It made a world of difference: servos are very responsive now. UDP is OK for my use case: I can send the state of the whole system in each message and lost messages or messages arriving out of order are not a big problem.

Sorry for the trouble.
